Add packwiz changelog generation script

exefer opened the pull request.

exefer commented:
Thinking of creating a subfolder for Bun scripts and moving `utils/index.js` to `<subfolder>/utils.js`. It's odd having JS files mixed with BAT, Bash, and PowerShell scripts. Also, if adding packages is needed in the future (even though the only script there is now doesn't require any), having a `package.json` in the Bun scripts folder is more ideal.

exefer commented:
Ok, the script is now fully functional with a couple of extra features compared to the original one 👀
I rebased everything into 3 commits that include commits spanning three days.
Do you also want to squash the testing commits you were making with Packwiz @WhitePhant0m?

EDIT: Actually just one commit - I made more changes after those 3, so the git history wasn't clean. Made a single commit instead.

exefer commented on automation/scripts/generate-changelog.js:
```diff
@@ -0,0 +1,272 @@
+  gitRepoPath: path.resolve(import.meta.dir, "..", ".."),
+  packVersion: null,
+  oldPackVersion: null,
+  fileId: "123456", // Do not change this
```
I think it would be better if the script prompted for a CF upload file ID when run by a human, instead of changing the file each time. What do you think?

WhitePhant0m replied:
Hmm, *maybe*?

WhitePhant0m commented:
> Thinking of creating a subfolder for Bun scripts and moving `utils/index.js` to `<subfolder>/utils.js`. It's odd having JS files mixed with BAT, Bash, and PowerShell scripts. Also, if adding packages is needed in the future (even though the only script there is now doesn't require any), having a `package.json` in the Bun scripts folder is more ideal.

Sounds good.

> Ok, the script is now fully functional with a couple of extra features compared to the original one 👀 I rebased everything into 3 commits that include commits spanning three days. Do you also want to squash the testing commits you were making with Packwiz @WhitePhant0m?

Nice.

As for the squashing, eh, not really sure if necessary, I won't fully merge this fork into the original, I'll be picking certain things from here to put there.

Mostly because I forgot that my dev env which I based this on has a "few" extra mods, and also missing some(mostly client side stuff that caused issues in dev only) and adding/removing all them is gonna be a pain as I don't remember all of them. :p

exefer commented:
Got it!

exefer commented:
Added tag support since tags are starting to be used now🏷️

WhitePhant0m commented:
ugh, shouldn't have synced the fork D:

exefer commented:
I deleted an older version of the file in my first commit, I believe that's causing the conflict.

exefer commented:
I should be able to fix it by unstaging the file deletion, pulling the latest version of `craftoria_changelog.py`, and deleting it again. Then I'll force push the changes 🙂

WhitePhant0m commented:
I'll probably merge this afterwards.

Also changed my mind about fully merging this into the main repo, after merging this I'll probably move the things from this fork over to the main repo onto a separate branch.

WhitePhant0m committed:
rhino my beloved

#582 made client fail to load, as rhino doesn't like destructuring in `catch()` D:

WhitePhant0m committed:
removed dummy multiblocks, as MI-Tweaks now supports standalone batchcrafters

WhitePhant0m committed:
format

WhitePhant0m committed:
feat(recipe): Add Xychorium ores to MI quarry (Bronze Drill); Add Kivi to MI quarry (Copper Drill)

also refactors quarry recipes to use MIs built in KJS compat.

WhitePhant0m committed:
feat(recipe): Add Replica ore to MI quarry (Titanium Drill)

WhitePhant0m committed:
refactor(lang): Simplify category assignment for language entries in en_us.js

also fixes the categories not working with standalone mi-tweaks batchcrafters

WhitePhant0m committed:
feat(config): update JAMD configs to include all current(findable normally) ores

exefer committed:
refactor: replace Python changelog automation with Node.js/Bun implementation

- Replace Python changelog automation with Node.js/Bun implementation
- Add full Packwiz compatibility for changelog generation
- Fetch and display mod author names in mod lists
- Move build scripts to dedicated scripts/ directory
- Enforce conventional commits and improve code organization
- Fix changelog generation for Modrinth mods and empty changelogs
- Extract reusable functions
- Use Packwiz metadata TOML for pack version retrieval
- Add auto-detection for cutoff commit (last version bump or release tag), instead of using dates which are inaccurate

exefer committed:
chore: delete Python-based changelog script

exefer commented:
Ok, fixed the conflict

WhitePhant0m commented:
yoink
